src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java.g

```diff
@@ -254,6 +257,7 @@ typeDefinitionInternal[AST modifiers]
     | interfaceDefinition[#modifiers]
     | enumDefinition[#modifiers]
     | annotationDefinition[#modifiers]
+    | recordDeclaration[#modifiers]
```

Comment:
What do you think about naming this `recordDefinition` to fit in with `enumDefinition`, `annotationDefinition`, etc.? (If you think that makes sense, no need to rerun reports just for that renaming)

Reply:
+1 for `recordDefinition`. Records are basically classes.

Reply:
I was thinking the same. Done.